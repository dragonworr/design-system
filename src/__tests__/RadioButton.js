import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { RadioButton, theme } from '..'

describe('RadioButton', () => {
  test('Selected, rendering', () => {
    const json = renderer
      .create(
        <RadioButton
          theme={theme}
          isSelected
          radioButtonText="Option Selected"
        />
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('Not Selected, rendering', () => {
    const json = renderer
      .create(
        <RadioButton theme={theme} radioButtonText="Option Not Selected" />
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('Disabled, rendering', () => {
    const json = renderer
      .create(
        <RadioButton theme={theme} disabled radioButtonText="Option Disabled" />
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })
})

describe('RadioButton onClick', () => {
  const onClick = jest.fn()

  test('Disabled', () => {
    const wrapper = mount(
      <RadioButton
        theme={theme}
        onClick={onClick}
        disabled
        radioButtonText="Option Disabled"
      />
    )
    wrapper.simulate('click')
    expect(onClick).toHaveBeenCalledTimes(0)
  })

  test('Not Disabled', () => {
    const wrapper = mount(
      <RadioButton
        theme={theme}
        onClick={onClick}
        radioButtonText="Option Not Selected"
      />
    )
    wrapper.simulate('click')
    expect(onClick).toHaveBeenCalled()
  })
})
