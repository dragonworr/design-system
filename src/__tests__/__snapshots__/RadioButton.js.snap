// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Disabled, rendering 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: borderGray;
}

.c0 {
  position: relative;
  cursor: pointer;
  font-weight: 400;
}

.c1 {
  position: absolute;
  height: 24px;
  top: -2px;
}

.c3 {
  padding-left: 32px;
}

<label
  className="c0"
  disabled={true}
  onClick={[Function]}
>
  <span
    className="c1"
  >
    <svg
      className="c2"
      color="borderGray"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12,2C6.5,2,2,6.5,2,12s4.5,10,10,10s10-4.5,10-10S17.5,2,12,2z M12,20c-4.4,0-8-3.6-8-8
      	s3.6-8,8-8s8,3.6,8,8S16.4,20,12,20z"
      />
    </svg>
  </span>
  <span
    className="c3"
  >
    Option Disabled
  </span>
</label>
`;

exports[`RadioButton Not Selected, rendering 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: borderGray;
}

.c0 {
  position: relative;
  cursor: pointer;
  font-weight: 400;
}

.c0:hover {
  color: #007aff;
}

.c0:hover svg {
  fill: #007aff;
}

.c1 {
  position: absolute;
  height: 24px;
  top: -2px;
}

.c3 {
  padding-left: 32px;
}

<label
  className="c0"
  disabled={undefined}
  onClick={[Function]}
>
  <span
    className="c1"
  >
    <svg
      className="c2"
      color="borderGray"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12,2C6.5,2,2,6.5,2,12s4.5,10,10,10s10-4.5,10-10S17.5,2,12,2z M12,20c-4.4,0-8-3.6-8-8
      	s3.6-8,8-8s8,3.6,8,8S16.4,20,12,20z"
      />
    </svg>
  </span>
  <span
    className="c3"
  >
    Option Not Selected
  </span>
</label>
`;

exports[`RadioButton Selected, rendering 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: blue;
}

.c0 {
  position: relative;
  cursor: pointer;
  font-weight: 600;
}

.c1 {
  position: absolute;
  height: 24px;
  top: -2px;
}

.c3 {
  padding-left: 32px;
}

<label
  className="c0"
  disabled={undefined}
  onClick={[Function]}
>
  <span
    className="c1"
  >
    <svg
      className="c2"
      color="blue"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12,7c-2.8,0-5,2.2-5,5s2.2,5,5,5s5-2.2,5-5S14.8,7,12,7z M12,2C6.5,2,2,6.5,2,12s4.5,10,10,10
      	s10-4.5,10-10S17.5,2,12,2z M12,20c-4.4,0-8-3.6-8-8s3.6-8,8-8s8,3.6,8,8S16.4,20,12,20z"
      />
    </svg>
  </span>
  <span
    className="c3"
  >
    Option Selected
  </span>
</label>
`;
